userguide/tutorials/invoice_examples.adoc

```diff
@@ -0,0 +1,132 @@
+= Invoice Examples
+
+This document provides examples of different invoice scenarios. 
+
+== Prerequisites
+
+Ensure that you have gone through the https://docs.killbill.io/latest/userguide_subscription.html#components-invoicing[_Subscription User Guide_] and are familiar with the basic invoicing concepts.
+
+
+== Invoice Terminology - work in progress
+
+Whenever there is a change in the system (a recurring subscription billing, an external charge, a change in plan, and so on), the system generates an invoice. An invoice consists of one or more invoice items of different types. In addition, an invoice consists of several components like `balance`, `chargedAmount`, `paidAmount` and so on. Let us first understand some of these terms. 
+
+
+* *Invoice Item*: Represents an amount corresponding to a type of charge, credit or adjustment. The different types of invoice items are `RECURRING`, `FIXED`, `EXTERNAL_CHARGE`, `USAGE`, `TAX`, `ITEM_ADJ`, `CREDIT_ADJ`, `REPAIR_ADJ` and `CBA_ADJ` as  explained in the https://docs.killbill.io/latest/userguide_subscription.html#components-invoicing[subscription billing] document. 
+
+* *Invoice Payment*: Represents a payment corresponding to an invoice. It has a `type` and an `amount`. The different types of invoice payments are `ATTEMPT` (represents successful payment), `CHARGED_BACK` (represents a chargeback) and `REFUND` (represents a refund) 
+
+
+* *chargedAmount*: The amount that the customer is charged. It is computed by adding up the amounts on invoice items of type `FIXED`, `RECURRING`, `EXTERNAL_CHARGE`, `REPAIR_ADJ`, `ITEM_ADJ`, `USAGE` and `TAX`. So, `chargedAmount=FIXED + RECURRING + EXTERNAL_CHARGE + REPAIR_ADJ + ITEM_ADJ + USAGE + TAX`  
+//TODO, add about including CREDIT_ADJ in case of DRAFT invoice?
```

Comment:
Yes, we definitely need to explain this (I would not talk about Draft though, this is just an artifact of how we can end up the Invoice credit scenario). Again, there are 2 scenarios, I'd like to use the following terminology:
* Invoice credit : an invoice with just a `CREDIT_ADJ` + `CBA_ADJ`
* Invoice adjustment : an invoice on which we added a `CREDIT_ADJ` -- it needed to be in Draft to do this, but not important; the point is that the `CREDIT_ADJ` is used to do an adjustment at the invoice level.

What I suggest is to claim that `chargedAmount` = Sum (`FIXED + RECURRING + EXTERNAL_CHARGE + REPAIR_ADJ + ITEM_ADJ + USAGE + TAX+ CREDIT_ADJ ` ), but then explain the use case of Invoice credit, and for which `CREDIT_ADJ` is not taken into account (otherwise the amount would end up negative).

Reply:
Also, try to present formulas like look more mathematical, e.g chargedAmount = Sum (FIXED + RECURRING + EXTERNAL_CHARGE + REPAIR_ADJ + ITEM_ADJ + USAGE + TAX+ CREDIT_ADJ ), I am not sure what the best way is, if you have some thoughts - I like the fact that you explain first and then provide an arithmetic definition.

-> The goal is that we don't need to read the code and we have something solid that provides all the details of how these amounts were computed.

Reply:
I've written about the `invoice credit` and `invoice adjustment` scenarios from what I understood based on this explanation.  I've also tried to present the `chargedAmount` and `invoice balance` as mathematical computations so they are easy to understand.